skipper_test.go

```diff
@@ -38,6 +39,45 @@ const (
 	listenTimeout = 9 * listenDelay
 )
 
+type protocol int
+
+const (
+	protoHTTP protocol = iota
+	protoHTTPS
+	protoH2C
+)
+
+func (p protocol) scheme() string {
+	return [...]string{"http", "https", "http"}[p]
+}
+
+func (p protocol) newClient() *http.Client {
+	switch p {
+	case protoHTTP:
+		return &http.Client{}
+	case protoHTTPS:
+		return &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
+				},
+			},
+		}
+	case protoH2C:
+		return &http.Client{
+			Transport: &http2.Transport{
+				// allow http scheme
+				AllowHTTP: true,
```

Comment:
I guess we should do the same for skipper's net.Client.

Reply:
Need to check if it continues to support http1 though. Otherwise we may think either about supporting h2c scheme for endpoints like `h2c://192.168.0.1` or have a filter that'd configure backend transport/roundtripper.

Reply:
That would be even better to support a scheme